File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int low = 0;
        int high = height.size()-1;


        int Area = 0;
        while(low < high){
            int mini = min(height[low],height[high]);
            int currArea = mini * (high - low );
            Area = max(currArea,Area);

            if(height[low] < height[high]){
                low ++;
            }
            else high--;

        }
        return  Area;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int low =0 , high = height.length-1;
        int maxArea = 0;

        while(low < high){
            int mini = Math.min(height[low],height[high]);
            int area = mini * (high - low);
            maxArea = Math.max(maxArea,area);

            if(height[low] < height[high]) low++;
            else high--;
        }
        return maxArea;
    }
}